packages/devtools_app/lib/src/debugger/flutter/debugger_controller.dart

```diff
@@ -0,0 +1,314 @@
+          await _service.getObject(isolateRef.id, scriptRef.id);
+    }
+
+    return _scriptCache[scriptRef.id];
+  }
+
+  SourcePosition calculatePosition(Script script, int tokenPos) {
+    final List<List<int>> table = script.tokenPosTable;
+    if (table == null) {
+      return null;
+    }
+
+    for (List<int> row in table) {
+      if (row == null || row.isEmpty) {
+        continue;
+      }
+      final int line = row.elementAt(0);
+      int index = 1;
+
+      while (index < row.length - 1) {
+        if (row.elementAt(index) == tokenPos) {
+          return SourcePosition(line, row.elementAt(index + 1));
+        }
+        index += 2;
+      }
+    }
+
+    return null;
+  }
+
+  String commonScriptPrefix;
+  LibraryRef rootLib;
+
+  void setRootLib(LibraryRef rootLib) {
+    this.rootLib = rootLib;
+
+    String scriptPrefix = rootLib.uri;
+    if (scriptPrefix.startsWith('package:')) {
+      scriptPrefix = scriptPrefix.substring(0, scriptPrefix.indexOf('/') + 1);
+    } else if (scriptPrefix.contains('/lib/')) {
+      scriptPrefix =
+          scriptPrefix.substring(0, scriptPrefix.lastIndexOf('/lib/'));
+      if (scriptPrefix.contains('/')) {
+        scriptPrefix =
+            scriptPrefix.substring(0, scriptPrefix.lastIndexOf('/') + 1);
+      }
+    } else if (scriptPrefix.contains('/bin/')) {
+      scriptPrefix =
+          scriptPrefix.substring(0, scriptPrefix.lastIndexOf('/bin/'));
+      if (scriptPrefix.contains('/')) {
+        scriptPrefix =
+            scriptPrefix.substring(0, scriptPrefix.lastIndexOf('/') + 1);
+      }
+    } else if (scriptPrefix.contains('/test/')) {
+      scriptPrefix =
+          scriptPrefix.substring(0, scriptPrefix.lastIndexOf('/test/'));
+      if (scriptPrefix.contains('/')) {
+        scriptPrefix =
+            scriptPrefix.substring(0, scriptPrefix.lastIndexOf('/') + 1);
+      }
+    } else {
+      scriptPrefix = null;
+    }
+
+    commonScriptPrefix = scriptPrefix;
+  }
+
+  int getLineNumber(Script script, dynamic location) {
+    if (script == null || location == null) {
+      return null;
+    }
+    if (location is UnresolvedSourceLocation && location.line != null) {
+      return location.line;
+    } else if (location is SourceLocation) {
+      return calculatePosition(script, location.tokenPos)?.line;
+    }
+    throw Exception(
+      '$location should be a $UnresolvedSourceLocation or a $SourceLocation',
+    );
+  }
+
+  String getShortScriptName(String uri) {
+    if (commonScriptPrefix == null) {
+      return uri;
+    }
+
+    if (!uri.startsWith(commonScriptPrefix)) {
+      return uri;
+    }
+
+    if (commonScriptPrefix.startsWith('package:')) {
+      return uri.substring('package:'.length);
+    } else {
+      return uri.substring(commonScriptPrefix.length);
+    }
+  }
+
+  void updateFrom(Isolate isolate) {
+    _breakpoints.value = isolate.breakpoints;
+  }
+}
+
+class SourcePosition {
+  SourcePosition(this.line, [this.column]);
```

Comment:
nit: make line a required named parameter  so it is more obvious the source positions are line, column

Reply:
Done.